impl/src/settings.rs

```diff
@@ -346,6 +347,11 @@ fn validate_settings(settings: &mut RazeSettings) -> Result<(), RazeError> {
     settings.workspace_path.pop();
   }
 
+  if settings.genmode == GenMode::Unspecified {
+    eprintln!("WARNING: The [raze] setting `genmode` is unspecified. Not specifying `genmode` is deperacated. Please explicitly set it to either \"Remote\" or \"Vendored\"");
+    settings.genmode = GenMode::Vendored;
```

Comment:
No requested change, just a continuation of my comment on "unreachable!".

You used "unreachable!" in that other code because of this code here, but I think that this code is too far removed from that code, and there's no sure guarantee that someone won't change this code and suddenly make that code reachable. Hence, my preference for different handling of this case.

---

Actually, on further inspection, it turns out that current code *definitely* could reach that other code. `validate_settings` is only called during `load_settings`. CrateSettings is actually a public struct, and therefore its fields could have any of their possible values.

Reply:
What should happen in this case?

Reply:
I think this code here should stay the same. Basically, everywhere else I'm saying "don't rely on that code!".

Reply:
Hey, I think I addressed everything in your other comments which is why I just marked them as resolved. Let me know what you think